### PR TITLE
refactor client folder layout

### PR DESCRIPTION
**Pull request by memsharded**

### CODE REVIEW

File: conans/__init__.py [hunk 1]
```diff
@@ -3,14 +3,13 @@
...
-from conans.client.cmake import CMake
-from conans.client.meson import Meson
-from conans.client.gcc import GCC
-from conans.client.configure_environment import ConfigureEnvironment
-from conans.client.configure_build_environment import (AutoToolsBuildEnvironment, VisualStudioBuildEnvironment)
+from conans.client.build.cmake import CMake
+from conans.client.build.meson import Meson
+from conans.client.build.gcc import GCC
+from conans.client.build.configure_environment import ConfigureEnvironment
+from conans.client.build.configure_build_environment import (AutoToolsBuildEnvironment, VisualStudioBuildEnvironment)
```
Comment (lasote) [inline conans/__init__.py]: Split in two.

Comment (memsharded) [reply conans/__init__.py]: What do you mean?

**Comment by lasote** [comment]
- Different folder with the command related collaborators? export, importer, new, new_ci, packager, package_copier, package_tester, remover, uploader...?
- output folder for output, printer, conan_command_output? (maybe a package output can be conflictive in the refactor)

File: conans/__init__.py (hunk 1 above)
Comment (lasote) [reply conans/__init__.py]: One for ``visual_studio`` and different one for ``autotools``

**Pull request merged**